### ISSUE TITLE
buildManifest 404s on random occasions, breaking the site

### ISSUE THREAD

**Issue by MitchEff** [vercel/next.js#25476]
### What version of Next.js are you using?

10.0.1

### What version of Node.js are you using?

12

### What browser are you using?

N/A

### What operating system are you using?

N/A

### How are you deploying your application?

Vercel

### Describe the Bug

When deployed to Vercel, around 1 in 10 deployments cannot open any pages except the first, as buildManifest.js returns a 404. When redeployed it often fixes it, but sometimes doesn't so I need to redeploy it again until it does. 

### Expected Behavior

I expect buildManifest.js to not 404 and the site to function correctly. 

### To Reproduce

Screenshot of issue is below. It seems completely random, so I'm unclear how to repro it. It's a stock-standard NextJS SSG site with a basic Vercel deployment. Nothing at all fancy going on. 

<img width="1680" alt="Screen Shot 2021-05-25 at 12 02 24 pm" src="https://user-images.githubusercontent.com/34614502/119585546-3e5c7b80-be0e-11eb-959b-ac1e81dbe974.png">

**Comment by MitchEff** [comment]
Is nobody else experiencing this? 

It's just happened again and completely breaks any navigation. I suspect Vercel is at fault as so far we've just had bug after bug with them, although I've seen some similar issues mentioned here. No idea how to address this.

**Comment by dan-b-hu** [comment]
Yes, we're randomly having this bug also. We've just updated to Next 10.2.3 and have done 6 deployments with no issues...

**Comment by timneutkens** [comment]
This was solved somewhere in 10.x, I'd recommend upgrading to Next.js 11 though to get all the latest performance improvements.

**Comment by MoSattler** [comment]
We're still experiencing the issue on 11.1

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
